packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-slug.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { safeDecodeURIComponent, filterURLForDisplay } from '@wordpress/url';
+import { useState, useMemo } from '@wordpress/element';
+import { __experimentalInspectorPopoverHeader as InspectorPopoverHeader } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import {
+	TextControl,
+	ExternalLink,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	Dropdown,
+	Button,
+} from '@wordpress/components';
+import { useEntityRecord, store as coreStore } from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import useEditedEntityRecord from '../../use-edited-entity-record';
+
+export const PERMALINK_POSTNAME_REGEX = /%(?:postname|pagename)%/;
+
+export default function PageSlug( { postType, postId } ) {
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const { editEntityRecord } = useDispatch( coreStore );
+	const { record, isLoaded } = useEditedEntityRecord( postType, postId );
+	const { record: savedRecord } = useEntityRecord( postType, postId );
+	const [ popoverAnchor, setPopoverAnchor ] = useState( null );
+	const isEditable =
+		PERMALINK_POSTNAME_REGEX.test( record?.permalink_template ) &&
+		record?._links?.[ 'wp:action-publish' ];
+	const viewPostLabel = useSelect(
+		( select ) => {
+			const postTypeObject = select( coreStore ).getPostType( postType );
+			return postTypeObject?.labels.view_item;
+		},
+		[ postType ]
+	);
+	const popoverProps = useMemo(
+		() => ( {
+			// Anchor the popover to the middle of the entire row so that it doesn't
+			// move around when the label changes.
+			anchor: popoverAnchor,
+			'aria-label': __( 'Change slug' ),
+			placement: 'bottom-end',
+		} ),
+		[ popoverAnchor ]
+	);
+	if ( ! isLoaded ) {
+		return null;
+	}
+	const onSlugChange = async ( newValue ) => {
+		try {
+			await editEntityRecord( 'postType', postType, postId, {
+				slug: newValue,
+			} );
```

Comment:
IIRC, the `editEntityRecord` shouldn't throw an error. We should also sanitize the new value. See `cleanForSlug`.

Reply:
Good catch here! I updated the code.